superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx

```diff
@@ -144,6 +150,7 @@ const FilterBar: React.FC<FiltersBarProps> = ({
 
   const [filtersInScope] = useSelectFiltersInScope(nativeFilterValues);
 
+  const dataMaskSelectedRef = useRef(dataMaskSelected);
```

Comment:
dataMaskSelectedRef only refers the first `dataMaskSelected` value only.
You should add the latest dataMaskSelected sync logic.
```suggestion
  const dataMaskSelectedRef = useRef(dataMaskSelected);
  dataMaskSelectedRef.current = dataMaskSelected;
```